feathr_project/feathr/anchor.py

```diff
@@ -27,6 +29,11 @@ def __init__(self,
     def validate_features(self):
         """Validate that anchor is non-empty and all its features share the same key"""
         assert len(self.features) > 0
+        if self.source != INPUT_CONTEXT:
+            for feature in self.features:
+                if feature.key == [DUMMY_KEY]:
+                    raise RuntimeError(f"For Non INPUT_CONTEXT(PASSTHROUGH) anchors, key of feature {feature.name} "
+                                       f"should be explicitly specified and DUMMY_KEY should not be used.")
```

Comment:
If input source is not INPUT_CONTEXT, key of feature {feature.name} should be explicitly specified and not left blank.

Maybe something like above? Since we don't talk about DUMMY KEY or PASSTHROUGH